images/dcos/root/etc/nginx/conf.d/default.conf.template

```diff
@@ -11,6 +19,25 @@ server {
 
     location / {
         proxy_pass <APISERVER>;
+        proxy_ssl_verify off;
+
+        # Authenticate
+        proxy_set_header Authorization "Bearer <TOKEN>";
+
+        # Don't try to be intelligent, only proxy
+        proxy_buffering off;
+        proxy_redirect off;
+        proxy_next_upstream off;
+
+        # Set long timeout
+        proxy_connect_timeout 159s;
```

Comment:
There is no heartbeating in kubectl-exec and other commands. So the connection is broken eventually without traffic. There are open GH issues upstream (kubernetes/kubernetes#18552, kubernetes/kubernetes#12179). I think I took the same values as the nginx in the GCE salt scripts. But these do not fix the actual problem.